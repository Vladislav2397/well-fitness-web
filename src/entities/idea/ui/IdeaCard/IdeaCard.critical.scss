.idea-card {
    min-width: 300px;
    min-height: 160px;
    aspect-ratio: 300 / 160;

    background-color: #ccc;

    $container: 1200 - 20 - 20;
    $count: 4;

    @media (min-width: $tablet) {
        min-height: 400px;

        &--vertical {
            min-width: 275px;
            flex-basis: 275px;
            aspect-ratio: 275 / 400;
        }

        &--horizontal {
            min-width: 575px;
            flex-basis: 575px;
            aspect-ratio: 575 / 400;
        }
    }
}
