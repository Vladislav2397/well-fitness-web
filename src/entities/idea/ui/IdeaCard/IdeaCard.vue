<template lang="pug">

.b-idea-card
    img(
        :src="idea.image.src"
        :alt="idea.image.alt"
    )

</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

export type IdeaCardProps = Pick<IdeaCard, 'idea'> &
    Partial<Pick<IdeaCard, 'orientation'>>

@Component
export default class IdeaCard extends Vue {
    @Prop() readonly idea!: { image: { src: string; alt: string } }
    @Prop({ default: 'vertical' }) readonly orientation!:
        | 'vertical'
        | 'horizontal'
}
</script>

<!--<style lang="scss" src="./IdeaCard.critical.scss" />-->
<!--<style lang="scss" src="./IdeaCard.main.scss" />-->
