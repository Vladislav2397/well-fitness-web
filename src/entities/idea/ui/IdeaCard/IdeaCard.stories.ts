import { Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import IdeaCard, { type IdeaCardProps } from './IdeaCard.vue'
import { defineControls } from '@/shared/lib/stories'

export default {
    title: 'undefined-path/IdeaCard',
    component: IdeaCard,
    argTypes: defineControls({
        //
    }),
}

const Template: Story<IdeaCardProps> = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { IdeaCard },
    methods: {
        handle: action('clicked'),
    },
    template: `<IdeaCard v-bind="$props" @click="handle" />`,
})

export const Default = Template.bind({})
Default.args = {
    //
}
