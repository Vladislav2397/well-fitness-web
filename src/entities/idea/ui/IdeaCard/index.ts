import IdeaCard from './IdeaCard.vue'

export { IdeaCard }
