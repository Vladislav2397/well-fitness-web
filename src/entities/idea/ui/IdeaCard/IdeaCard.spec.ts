import { shallowMount } from '@vue/test-utils'
import IdeaCard /*, { type IdeaCardProps }*/ from './IdeaCard.vue'

describe('IdeaCard', () => {
    test('snapshot', () => {
        const wrapper = shallowMount(IdeaCard, {
            propsData: {
                idea: {
                    image: {
                        src: 'path/to/image',
                        alt: 'image',
                    },
                },
            } /* as IdeaCardProps*/,
        })
        expect(wrapper).toMatchSnapshot()
    })
})
