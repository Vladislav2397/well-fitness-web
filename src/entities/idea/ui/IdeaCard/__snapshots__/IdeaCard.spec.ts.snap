// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard snapshot 1`] = `<div class="b-idea-card"><img src="path/to/image" alt="image"></div>`;
